synapse/api/auth.py

```diff
@@ -513,69 +508,38 @@ def is_server_admin(self, user):
         """
         return self.store.is_server_admin(user)
 
-    @defer.inlineCallbacks
```

Comment:
turning this from a deferred-returning function to an immediate one without updating the call sites feels like it might bite us later? Better to `return defer.suceed()` or add a `yield defer.succeed()` ?

Not that bothered though. We'll probably catch it when we update the call sites...